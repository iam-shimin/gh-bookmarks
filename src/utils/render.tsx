import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import store from 'store';

// Test render

export default function renderWithStoreAndRouter(element: React.ReactElement) {
	return render(
		<MemoryRouter>
			<Provider store={store}>{element}</Provider>
		</MemoryRouter>
	);
}