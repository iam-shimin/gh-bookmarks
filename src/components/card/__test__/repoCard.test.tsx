import React from 'react';
import render from 'utils/render';
import RepoCard from '../userCard';

describe('<RepoCard />', () => {
	test('match snapshot', () => {
		const { asFragment } = render(<RepoCard data={{
			avatar_url: 'avatar_url',
			login: 'login',
			html_url: 'html_url',
			id: 'id'
		}} />);
		expect(asFragment()).toMatchSnapshot();
	});
})