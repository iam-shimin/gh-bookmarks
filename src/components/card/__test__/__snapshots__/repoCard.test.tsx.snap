// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoCard /> match snapshot 1`] = `
<DocumentFragment>
  <div
    class="card repo-card"
  >
    <a
      href="https://github.com/facebook/react"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        class="title"
        role="heading"
      >
        facebook/react/17-latest
      </span>
    </a>
    <small>
      facebook/react
    </small>
    <p>
      A declarative, efficient, and flexible JavaScript library for building user interfaces.
    </p>
    <div
      class="mt-1"
    >
      <button
        class="ml-1 add-bmk-btn btn-sep btn btn-primary"
        type="button"
      >
        Add bookmark
      </button>
    </div>
  </div>
</DocumentFragment>
`;
